src/python/pants/option/options_bootstrapper_test.py

```diff
@@ -142,67 +162,66 @@ def test_bootstrapped_options_ignore_irrelevant_env(self):
-      opts_single_config.register('', '--pants-config-files', type=list)
-      opts_single_config.register('', '--config-override', type=list)
-
-      opts_single_config.register('compile.apt', '--worker-count')
-      opts_single_config.register('fruit', '--apple')
-
-      self.assertEqual('1', opts_single_config.for_scope('compile.apt').worker_count)
-      self.assertEqual('red', opts_single_config.for_scope('fruit').apple)
-
-      with temporary_file(binary_mode=False) as fp2:
-        fp2.write(dedent("""
-        [compile.apt]
-        worker_count: 2
-        """))
-        fp2.close()
-
-        bootstrapper_double_config = create_options_bootstrapper(fp.name, fp2.name)
-
-        opts_double_config = bootstrapper_double_config.get_full_options(known_scope_infos=[
-          ScopeInfo('', ScopeInfo.GLOBAL),
-          ScopeInfo('compile.apt', ScopeInfo.TASK),
-          ScopeInfo('fruit', ScopeInfo.TASK),
-        ])
-        # So we don't choke on these on the cmd line.
-        opts_double_config.register('', '--pants-config-files', type=list)
-        opts_double_config.register('', '--config-override', type=list)
-        opts_double_config.register('compile.apt', '--worker-count')
-        opts_double_config.register('fruit', '--apple')
-
-        self.assertEqual('2', opts_double_config.for_scope('compile.apt').worker_count)
-        self.assertEqual('red', opts_double_config.for_scope('fruit').apple)
-
-  def test_create_bootstrapped_multiple_pants_config_files(self):
-    def create_options_bootstrapper(*config_paths):
-      return OptionsBootstrapper.create(args=[f'--pants-config-files={cp}' for cp in config_paths])
+      options.register('', '--pants-config-files', type=list)
+      options.register('', '--config-override', type=list)
+      options.register('compile.apt', '--worker-count')
+      options.register('fruit', '--apple')
+
+      self.assertEqual(str(expected_worker_count), options.for_scope('compile.apt').worker_count)
+      self.assertEqual('red', options.for_scope('fruit').apple)
+
+    with temporary_file(binary_mode=False) as fp1, temporary_file(binary_mode=False) as fp2:
+      fp1.write(
+        dedent(
+          """\
+         [compile.apt]
+         worker_count: 1
+  
+         [fruit]
+         apple: red
+         """
+        )
+      )
+      fp2.write(
+        dedent(
+          """\
+          [compile.apt]
+          worker_count: 2
+          """
+        )
+      )
+      fp1.close()
+      fp2.close()
 
-    self.do_test_create_bootstrapped_multiple_config(create_options_bootstrapper)
+      assert_config_read_correctly(
+        create_options_bootstrapper(fp1.name), expected_worker_count=1,
+      )
+      assert_config_read_correctly(
+        create_options_bootstrapper(fp1.name, fp2.name), expected_worker_count=2,
+      )
+      assert_config_read_correctly(
+        create_options_bootstrapper(fp2.name, fp1.name), expected_worker_count=1,
+      )
```

Comment:
We didn't test this edge case previously. The refactor makes it easy to add the check.